cpp/Esri/ArcGISRuntime/Toolkit/PopupViewController.h

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ *  Copyright 2012-2020 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#ifndef ESRI_ARCGISRUNTIME_TOOLKIT_POPUPVIEWCONTROLLER_H
+#define ESRI_ARCGISRUNTIME_TOOLKIT_POPUPVIEWCONTROLLER_H
+
+// ArcGISRuntime headers
+#include <PopupManager.h>
+
+// Qt headers
+#include <QAbstractListModel>
+#include <QObject>
+
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+
+namespace Toolkit
+{
+
+/*!
+ * \brief In MVC architecture, this is the controller for the corresponding
+ * PopupView.
+ * 
+ * This controller is a thin wrapper around a PopupManager. It re-exposes some
+ * PopupManager properties, including the number of total rows to render as a 
+ * property.
+ */
+class PopupViewController : public QObject
+{
+  Q_OBJECT
+  Q_PROPERTY(PopupManager* popupManager READ popupManager WRITE setPopupManager NOTIFY popupManagerChanged)
+  Q_PROPERTY(QAbstractListModel* displayFields READ displayFields NOTIFY popupManagerChanged)
+  Q_PROPERTY(int fieldCount READ fieldCount_ NOTIFY fieldCountChanged)
```

Comment:
just curious why we need the trailing "_"? Is there a name clash?

Reply:
Going on previous comments that private member functions should have trailing `_`, this now reads from a private member function.